--- lmsquery.py.orig	2020-05-29 14:50:32.895983000 +0300
+++ lmsquery.py	2022-11-04 20:25:29.949000000 +0200
@@ -5,13 +5,15 @@
 
 from . import const
 
-
 class LMSQuery(object):
-    def __init__(self, host=const.LMS_HOST, port=const.LMS_PORT, player_id=""):
+    def __init__(self, host=const.LMS_HOST, port=const.LMS_PORT, player_id="", timeout=2):
         self.host = host
         self.port = port
         self.server_url = "http://%s:%s/jsonrpc.js" % (self.host, self.port)
+        self.session = requests.Session()   # re-use connection
         self.player_id = player_id
+        self.timeout = timeout
+
 
 ###############################################################################
 # Generic query
@@ -19,9 +21,10 @@
     def query(self, player_id="", *args):
         params = json.dumps({'id': 1, 'method': 'slim.request',
                              'params': [player_id, list(args)]})
-        r = requests.post(self.server_url, params)
+        r = self.session.post(self.server_url, params, timeout=self.timeout)
         return json.loads(r.text)['result']
 
+
 ###############################################################################
 # Server commands
 ###############################################################################
